proposals/NNNN-optional-label-for-first-trailing-closure.md

```diff
@@ -0,0 +1,653 @@
+# Support use of an optional label for the first trailing closure
+
+* Proposal: [SE-NNNN](NNNN-optional-label-for-first-trailing-closure.md)
+* Authors: [Xiaodi Wu](https://github.com/xwu)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: [apple/swift#31666](https://github.com/apple/swift/pull/31666)
+
+<!--
+*During the review process, add the following fields as needed:*
+
+* Decision Notes: [Rationale](https://forums.swift.org/), [Additional Commentary](https://forums.swift.org/)
+* Bugs: [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN), [SR-MMMM](https://bugs.swift.org/browse/SR-MMMM)
+* Previous Revision: [1](https://github.com/apple/swift-evolution/blob/...commit-ID.../proposals/NNNN-filename.md)
+* Previous Proposal: [SE-XXXX](XXXX-filename.md)
+-->
+
+## Introduction
+
+This proposal extends trailing closure syntax by supporting the use of an
+__optional__ argument label for the first trailing closure. When paired with
+changes in tooling behavior, this feature would promote the alignment of
+declaration and use sites without source-breaking changes.
+
+<!--
+Swift-Evolution thread: [TBD](https://forums.swift.org/)
+-->
+
+## Motivation
+
+__Closure expressions__ in Swift have a lightweight syntax with several
+"optimizations" as described in _The Swift Programming Language_:
+
+> * Inferring parameter and return value types from context
+> * Implicit returns from single-expression closures
+> * Shorthand argument names
+> * Trailing closure syntax
+
+These optimizations allow users to write out only the most salient details
+explicitly at the point of use, promoting clarity when used judiciously. A
+caller can take advantage of some or all of these optimizations in a variety
+of combinations. For instance, one may use trailing closure syntax while
+explicitly naming the return type.
+
+<!--
+Take, for instance, the different ways of calling the following function:
+
+```swift
+func frobnicate(by: (String) -> Bool) { /* ... */ }
+
+frobnicate(by: { (value: String) -> Bool in
+  return value == "Hello"
+})
+
+// Inferring parameter type:
+frobnicate(by: { (value) -> Bool in
+  return value == "Hello"
+})
+frobnicate(by: { value -> Bool in
+  return value == "Hello"
+})
+
+// Inferring return type:
+frobnicate(by: { value in
+  return value == "Hello"
+})
+
+// Implicit return from single-expression closure
+// (now available for all functions):
+frobnicate(by: { value in
+  value == "Hello"
+})
+
+// Shorthand argument name:
+frobnicate(by: { $0 == "Hello" })
+
+// Trailing closure syntax:
+frobnicate() { $0 == "Hello" }
+frobnicate   { $0 == "Hello" }
+```
+
+These optimizations do not have to be used in only the progression outlined
+above. For instance, one may use trailing closure syntax while explicitly naming
+the return type.
+-->
+
+In Swift, function arguments can have labels prescribed by the author of the
+function that clarify the role of that argument. For instance, the mathematical
```

Comment:
`clarifies` will be more suitable than `clarify`

```suggestion
function that clarifies the role of that argument. For instance, the mathematical
```

Reply:
The verb “clarify” agrees with “labels.”